### ISSUE TITLE
checkJs: allow destructuring URL in JSDoc.

### ISSUE THREAD

**Issue by regseb** [microsoft/TypeScript#35826]
**TypeScript Version:**  3.7.4

**Search Terms:** checkJs, allowJs, JavaScript, JSDoc, TS8032, URL, destructuring, object.

**Code**

* testcase.js

```javascript
/**
 * Foo.
 *
 * @param {URL}    url      Bar.
 * @param {string} url.href Baz.
 * @return {string} Quz.
 */
function foo({ href }) {
    return href;
}
```

* tsconfig.json

```json
{
    "compilerOptions": {
        "allowJs": true,
        "noEmit":  true
    },
    "include": ["testcase.js"]
}
```

* `npx tsc --checkJs`

**Expected behavior:** allow destructuring URL in JSDoc.

**Actual behavior:**

```
testcase.js:5:20 - error TS8032: Qualified name 'url.href' is not allowed without a leading '@param {object} url'.

5  * @param {string} url.href Baz.
                     ~~~~~~~~
```

**Playground Link:**

* download and unzip [_testcase.zip_](https://github.com/microsoft/TypeScript/files/3995115/testcase.zip);
* execute: `npm install` and `npx tsc --checkJs`.

**Related Issues:** none.

**Comment by Manu1400** [comment]
The TypeScript Playground link is bad

**Comment by regseb** [comment]
@Manu1400 TypeScript Playground doesn't report error. So I attached a [zip](https://github.com/microsoft/TypeScript/files/3995115/testcase.zip) to reproduce the problem.

**Comment by DanielRosenwasser** [comment]
What exactly is the purpose of this code? `URL` already has a typed declaration of `href` elsewhere.

**Comment by regseb** [comment]
I misunderstood. I thought we should specify all destructuring in the JSDoc. But this isn't necessary for `URL` because TypeScript already knows the structure.

This code works:

```javascript
/**
 * Foo.
 *
 * @param {URL} url Bar.
 * @return {string} Quz.
 */
function foo({ href }) {
    return href;
}
```

Thank you for your response and sorry for this false issue.

**Comment by DanielRosenwasser** [comment]
No worries!
